sys/netinet/ip_carp.c

```diff
@@ -1254,7 +1280,7 @@ carp_master_down_locked(struct carp_softc *sc, const char *reason)
 
 	switch (sc->sc_state) {
 	case BACKUP:
-		carp_set_state(sc, MASTER, reason);
+		carp_set_state(sc, (V_carp_allow != 2) ? MASTER : BACKUP, reason);
```

Comment:
I don't think this is a proper place/way to check the V_carp_allow.  Even if V_carp_allow == 2, the lines below will send ARP/ND announces, which are odd to see while the node remains in a backup state.

Reply:
I've reworked this a bit. I think having V_carp_allow here is okay. What is issue with having it here?